File: core-algorithms/1_recursion/4_array_product_sum/array_product_sum.py
def array_product_sum(lis, depth=1):
    sum = 0
    for l in lis:
        if(isinstance(l, list)):
            sum = sum + array_product_sum(l, depth+1)
        else:
            sum += l 
    return sum * depth

print(array_product_sum([1, 2, [2, 3], 6, [[2, 3], 1], 7]))

File: core-algorithms/1_recursion/4_array_product_sum/array_product_sum.ts
const array_product_sum = (arr: any, depth = 1) => {
  let sum = 0;
  for (let i of arr) {
    if (Array.isArray(i)) {
      sum = sum + array_product_sum(i, depth + 1);
    } else {
      sum = sum + i;
    }
  }
  return sum * depth;
};

console.log(array_product_sum([1, 2, [2, 3], 6, [[2, 3], 1], 7]));
